clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -128,13 +127,6 @@ private void changeSubscription(Set<String> topicsToSubscribe) {
         if (!this.subscription.equals(topicsToSubscribe)) {
             this.subscription = topicsToSubscribe;
             this.groupSubscription.addAll(topicsToSubscribe);
-
-            // Remove any assigned partitions which are no longer subscribed to
-            for (Iterator<TopicPartition> it = assignment.keySet().iterator(); it.hasNext(); ) {
```

Comment:
What I'm looking for is test cases which would have caused this code deletion to fail. Perhaps the main case we're missing is one which asserts that the assignment remains the same after a subsequent call to `subscribe()`, and only changes when `assignFromSubscribed` is called. Glancing at the code, I'm also not sure we have a case which covers a subsequent call to `subscribe(Pattern)` as well.